contracts/DKVDaggerHashimoto.sol

```diff
@@ -43,11 +44,16 @@ contract DecentralizedKVDaggerHashimoto is DecentralizedKV {
         payable
         DecentralizedKV(_config.systemContract, 1 << _config.maxKvSizeBits, _startTime, _storageCost, _dcfFactor)
     {
+        /* Assumptions */
+        require(_config.shardSizeBits >= _config.maxKvSizeBits, "shardSize too small");
+        require(_config.randomChecks > 0, "At least one checkpoint needed");
+
         systemContract = _config.systemContract;
         shardSizeBits = _config.shardSizeBits;
         maxKvSizeBits = _config.maxKvSizeBits;
-        require(_config.shardSizeBits >= _config.maxKvSizeBits, "shardSize too small");
         shardEntryBits = _config.shardSizeBits - _config.maxKvSizeBits;
+        chunkLenBits = _config.maxKvSizeBits < DEFAULT_CHUNK_SIZE_BITS ?
```

Comment:
I think it may be better to move chunkLenBits to DecentralizedKV.sol and replace DEFAULT_CHUNK_SIZE_BITS?  It sounds to me that 0 chunkLenBits is invalid.

Reply:
It should keep aligned with _config.maxKvSizeButs, since there clearly exists a dependency in between. Unless we move maxKvSizeBits into DecentralizedKV.sol, otherwise it does not make too much sense.
I'd rather keep it here.

Reply:
Oh, what I mean is to replace DEFAULT_CHUNK_SIZE_BITS with an immutable chunkSizeBits, and put require(maxKvSizeBits >= chunkSizeBIts, "...") in the ctor.